TACC/Core-Portal
# task/TV3-79 Postits

## Overview

The goal for this task was to port over the postit functionality to v3

## Related

* [TV3-79](https://jira.tacc.utexas.edu/browse/TV3-79)

## Changes

- Using the new tapis commands for creating, deleting and updating postits 
- Updated the `Link` model; renamed `agave_uri` field to `tapis_uri`, added a new field for `expiration` which tracks how long a postit is valid for. The expiration date is displayed in the Link modal after a postit has been generated 
- Cleaned up and removed some unused arguments from functions

## Testing

1. Go to a My Data system 
2. Download a file and ensure it downloads correctly. 
3. Click on the Link button and generate a postit and ensure functionality works 
4. Try deleting postit and replacing link and ensure that functionality works as well
5. Try previewing a file by clicking on it and ensure preview works as intended

## UI

<img width="2048" alt="Screen Shot 2023-03-21 at 2 12 02 PM" src="https://user-images.githubusercontent.com/23081932/226716339-a3206816-0343-40e3-9d1d-0b581d8403ef.png">


## Notes

There are a couple things currently in progress: 

<s>
- Postit functionality isn't deployed to prod tapipy yet therefore using dev version of tapipy for now 
- I will be working on updating/fixing tests as well
</s>
